core/src/main/java/org/apache/iceberg/rest/BaseHTTPClient.java

```diff
@@ -77,6 +77,18 @@ public <T extends RESTResponse> T get(
     return execute(request, responseType, errorHandler, h -> {});
   }
 
+  @Override
+  public <T extends RESTResponse> T get(
```

Comment:
If we're adding this, would it be possible to push the other `get` implementation to the interface?

Reply:
Sorry for the late response.

Do you mean we could move that other `get` function from this class into the interface? We could have a "responseHeader-less" version in the interface that calls the version with the responseHeader param using a non-null `h -> {}` consumer, but I think that would be a behaviour change within the interface. Calling the "responseHeader-less" version would result in an `UnsupportedOperationException` from the default implementation of the other get function in the interface.

With the current approach I tried to follow how [this PR](https://github.com/apache/iceberg/pull/6578) from @nastra introduced the same for the `post` methods.